cowbird/permissions_synchronizer.py

```diff
@@ -269,43 +318,59 @@ def _find_matching_res(self, service_type: str, resource_nametype_path: str) ->
 
     @staticmethod
     def _create_res_data(target_segments: List[ConfigSegment],
-                         input_matched_groups: Dict[str, str],
+                         input_matched_groups: Collection[str],
                          ) -> List[ResourceSegment]:
         """
         Creates resource data, by replacing any tokens found in the segment names to their actual corresponding values.
         This data includes the name and type of each segments of a full resource path.
+        In the case where a regex is found in the target segment, the data will be formed using the same resource_type
+        for every match in the current segment.
 
         :param target_segments: List containing the name and type info of each segment of the target resource path.
         :param input_matched_groups:
         """
         res_data: List[ResourceSegment] = []
         for segment in target_segments:
-            matched_groups = re.match(NAMED_TOKEN_REGEX, segment["name"])
-            if matched_groups:
-                res_data.append({
-                    "resource_name": input_matched_groups[matched_groups.groups()[0]],
-                    "resource_type": segment["type"]
-                })
-            elif segment["name"] == MULTI_TOKEN:
-                multi_segments = input_matched_groups["multi_token"]
-                # Skip the segment if the multi_token matched 0 times, resulting in an empty string.
+            # Use the regex to create the res_data
+            if segment.get("regex") is not None:
+                regex = segment.get("regex")
+                matches = re.search(regex, input_matched_groups)
+                multi_segments = matches.group()
```

Comment:
In the target res key , yes since we need to be able to recreate the right resource data, and there is no way to know the type of the matched segment other than from the config. Therefore we can use the same type as what the regex matched and recreate "multiple segments" from a single one. ex: {thredds:birdhouse/testdata/xclim/cmip6} and {thredds:birdhouse/testdata/xclim/cmip6/file.nc} We could want to apply the permission on the directory itself or on the file:
`thredd_collection:
type: directory
regex: '(?<=:)[\w\/]+' `
(this would create any number of directory)
`thredd_item: 
`type: directory
regex: '(?<=:).*\/?(?=\/)' 
`type: file  
regex: '[^\/]+$'` 
(this would create any number of directory and a file )

Reply:
Ah! Very nice example. If that can be added to the example config, that would be great.